consensus/src/experimental/commit_phase.rs

```diff
@@ -343,6 +363,27 @@ impl CommitPhase {
         self.back_pressure.load(Ordering::SeqCst)
     }
 
+    pub async fn process_reset_event(
+        &mut self,
+        reset_event_callback: oneshot::Sender<ResetEventContentType>,
+    ) -> anyhow::Result<()> {
+        // reset the commit phase
+
+        // exhaust the commit channel
+        // we do not have to exhaust the commit message channel because inconsistent messages will be ignored
+        while self.commit_channel_recv.next().now_or_never().is_some() {}
+
+        // reset local block
+        self.blocks = None;
+
+        // activate the callback
+        reset_event_callback
+            .send(reset_event_content_new())
+            .map_err(|_| Error::ResetDropped)?;
```

Comment:
what would happen if the error happens? if we can't recover we'd better panic

Reply:
the host (execution phase) might drop. I think it is fine because whenever we drop the execution phase, we drop the commit phase as well.